src/peft/peft_model.py

```diff
@@ -123,7 +129,7 @@ def __init__(self, model: PreTrainedModel, peft_config: PeftConfig, adapter_name
         else:
             self._peft_config = None
             cls = PEFT_TYPE_TO_MODEL_MAPPING[peft_config.peft_type]
-            self.base_model = cls(model, {adapter_name: peft_config}, adapter_name)
+            self.base_model = cls(model, {adapter_name: peft_config}, adapter_name, self)
```

Comment:
I really want to avoid this, as this looks like a mixing of concerns. Surely, we can figure out a better way. Could you explain why this was needed?

Reply:
This was needed because in `XLoraModel.__init__` we load adapters into the `PeftModel` via `load_adapter` and for use by the XLoraClassifier. This way, automatic loading is achieved. However, all that matters is that we are able to get some reference to `PeftModel` into the constructor. 

Perhaps we could add some code after we call `cls(model...)`, and check if `self.base_model` is a `XLoraModel`. Then, we could call a `__post_init__` which would take the `PeftModel` self, and do the adapter loading. Would this be more elegant?

Reply:
I just pushed a commit ([c5cdfc3](https://github.com/huggingface/peft/pull/1491/commits/c5cdfc3d75c9eca8b2b4929662f03ab94a7c5ae0)) which is the latest one currently for easy reversion. Does this resolve the concern?